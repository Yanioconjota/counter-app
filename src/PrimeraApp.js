import React from 'react';
import PropTypes from 'prop-types';

const string = 'Esto es un string';
const numeros = 666;
const booleano = true;
const arr = ['🍲', '🍔', '🍟', '🍕', '🌭', '🥪'];
const obj = {
  nombre: 'James',
  apellido: 'Howlett',
  alias: 'Wolverine',
  edad: 'desconocida'
}
// const PrimeraApp = ({saludo}) es la desestructuración de las props
//const PrimeraApp = (props)
const PrimeraApp = ({
  saludo = 'Hola Insecto, esto es una prop default',
  otraPropiedad,
  subtitulo }) => {

  console.log({saludo});

  return (
    <>
      <h1>{saludo}</h1>
      <p>Otra propType tipo number: {otraPropiedad}</p>
      <p id="defaultProps">defaultProps: {subtitulo}</p>
      <p>Imprimir string: {string}</p>
      <p>Imprimir numeros: {numeros}</p>
      <p>Imprimir booleano: {booleano} no aparecen! 🙃</p>
      <p>Imprimir arrays: {arr}</p>
      <pre>{JSON.stringify( obj, null, 4)}</pre>
      <p>Párrafo común</p>
    </>
  )
    
}

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired,
  otraPropiedad: PropTypes.number.isRequired
}

PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtítulo!'
}

export default  PrimeraApp;

/*
import React, { Fragment } from 'react';
<Fragment>
  <h1>Hola Mundo</h1>
  <p>Párrafo</p>
</Fragment>
*/