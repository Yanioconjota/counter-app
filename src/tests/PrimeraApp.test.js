import React from 'react';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';
import '@testing-library/jest-dom';
// import { render } from '@testing-library/react';

describe('Pruebas en <PrimeraApp/>', () => {
  // test('Debe mostrar el mensaje: Hola, soy Goku y estás leyendo esto con mi voz!', () => {
  //   const saludo = 'Hola, soy Goku y estás leyendo esto con mi voz!';
  //   const otraPropiedad = 666;
  //   const { getByText } = render(<PrimeraApp saludo={saludo} otraPropiedad={otraPropiedad}/>);

  //   expect(getByText(saludo)).toBeInTheDocument();
  // });
  test('Debe mostrar <PrimeraApp/> Correctamente', () => {
    const saludo = 'Hola, soy Goku y estás leyendo esto con mi voz!';
    const otraPropiedad = 666;
    const wrapper = shallow(<PrimeraApp saludo={saludo} otraPropiedad={otraPropiedad}/>);
    expect(wrapper).toMatchSnapshot();
  });
  test('Debe mostrar el subtítulo enviado por props ', () => {
    const saludo = 'Hola, soy Goku y estás leyendo esto con mi voz!';
    const otraPropiedad = 666;
    const subTitulo = 'Soy un subtítulo!';

    const wrapper = shallow(<PrimeraApp
      saludo={saludo}
      otraPropiedad={otraPropiedad}
      subtitulo={subTitulo}
    />);

    const textoParrafo = wrapper.find('#defaultProps').text();
    console.log(textoParrafo);
    console.log(subTitulo);
    expect(textoParrafo).toBe(`defaultProps: ${subTitulo}`);
  })
  
});
