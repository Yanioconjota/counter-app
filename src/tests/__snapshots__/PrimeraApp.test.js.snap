// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> Debe mostrar <PrimeraApp/> Correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku y estás leyendo esto con mi voz!
  </h1>
  <p>
    Otra propType tipo number: 
    666
  </p>
  <p
    id="defaultProps"
  >
    defaultProps: 
    Soy un subtítulo!
  </p>
  <p>
    Imprimir string: 
    Esto es un string
  </p>
  <p>
    Imprimir numeros: 
    666
  </p>
  <p>
    Imprimir booleano: 
    <Component />
     no aparecen! 🙃
  </p>
  <p>
    Imprimir arrays: 
    🍲
    🍔
    🍟
    🍕
    🌭
    🥪
  </p>
  <pre>
    {
    "nombre": "James",
    "apellido": "Howlett",
    "alias": "Wolverine",
    "edad": "desconocida"
}
  </pre>
  <p>
    Párrafo común
  </p>
</Fragment>
`;
